### ISSUE TITLE
Component Sourcemaps using @import do not sourcemap correctly

### ISSUE THREAD

**Issue by narthur157** [angular/angular-cli#9271]
### Versions
```
_                      _                 ____ _     ___
/ \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
/ △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
/ ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
/_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
|___/

Angular CLI: 1.7.0-beta.0
Node: 6.11.1
OS: win32 x64
Angular: 5.2.1
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cdk: 5.1.0
@angular/cli: 1.7.0-beta.0
@angular/material: 5.1.0
@angular-devkit/build-optimizer: 0.0.38
@angular-devkit/core: 0.0.25
@angular-devkit/schematics: 0.0.48
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.10.0-beta.0
@schematics/angular: 0.1.13
@schematics/schematics: 0.0.13
typescript: 2.4.2
webpack: 3.10.0
```
### Repro steps
* Step 1 Clone https://github.com/narthur157/sourcemap-bug
* Step 2 npm install, then ng serve -sm
* Step 3 Inspect the red text ![image](https://user-images.githubusercontent.com/2042752/35105209-548c9a88-fc6b-11e7-8636-cb7911fc4054.png)
* Step 4 Notice how there is a `color: red` coming from `_theming.scss`
* Step 5: Observe `src/app/app.component.scss` (where this style comes from) 

```
@import '../variables.scss';

form > p {
    color: red;
}
```

Step 6: Try removing `@import '../variables.scss';`
Step 7: Notice how the sourcemap now correctly points to `app.component.scss`

### Observed behavior
Sourcemap points to the imports instead of the actual file.

### Desired behavior
Working sourcemap

### Mention any other details that might be useful (optional)

This may be an angular bug, but I'm not really sure and, in any case, it affects the user's of angular-cli. I've tried this with different versions of angular-cli with no difference in results. All other sourcemapping tests I tried seem to work, but this quite-common case fails quite often. 

Basically I don't have enough knowledge in how exactly the sourcemapping is done to know how to go about fixing this. Also I find the resulting URL odd: 
![image](https://user-images.githubusercontent.com/2042752/35105982-54008686-fc6d-11e7-943c-39f054a93936.png)

It appends the path to the material import on to the path to the folder of the correct style file

**Comment by alan-agius4** [comment]
Thanks for reporting this issue. This issue is now obsolete due to changes in the recent releases. Please update to the most recent Angular CLI version.

If the problem persists after upgrading, please open a new issue, provide a simple repository reproducing the problem, and describe the difference between the expected and current behavior.
